espnet2/bin/s2t_inference.py

```diff
@@ -566,17 +568,28 @@ def decode_long(
         if isinstance(speech, np.ndarray):
             speech = torch.tensor(speech)
 
-        assert (
-            speech.dim() == 1
-        ), f"speech must have one dimension, got size {speech.size()} instead"
+        if speech.dim() > 1:
+            assert (
+                speech.dim() == 2 and speech.size(1) == 1
+            ), f"speech of size {speech.size()} is not supported"
+            speech = speech.squeeze(1)  # (nsamples, 1) --> (nsamples,)
 
         utterances = []
         offset = 0
         text_prev = init_text
         while offset < len(speech):
             logging.info(f"Current start time in seconds: {offset / fs:.2f}")
-
             segment = speech[offset : offset + segment_len]
+            if (
+                offset + segment_len > len(speech)
```

Comment:
Do we need this condition? This is equivalent to `len(segment) < segment_len`, where `segment_len` is much larger than our threshold in the next line. So, I think this condition will be satisfied automatically if the next condition is true.

Reply:
That's true. Didn't notice that if `len(segment) < segment_len`, that's the last chunk.